free-1.md

```diff
@@ -10,41 +10,148 @@ _What need is met by your idea? Who are you helping? What is that person's
+CSound, effects must be implemented from filters and envelopes. At a more
+intuitive (for musicians) level of abstraction, primitive effects would include
+compressors and equalizers. These effects could be chained together with
+sources to alter the sound produced by the sources.
+
+A good tool for digital music synthesis should provide these primitives, and
+should allow them to be composed by chaining in serial and in parallel. This
+already sounds like a language -- we have vocabulary (sources and effects) and
+fluency (the method for chaining).
 
 ### Why you?
 _What excites you about this idea? How did you come up with it?_
 
+I have been interested in making music of all kinds for years. I took Music 88 last year, and while I enjoyed learning CSound and using it to make music, I also felt frustrated that it is so difficult to use.
+
+I was discussing potential project ideas with Daniel Zhang when the conversation turned to Melodica, and we both felt that it would be fun to make a music language like that. It seemed natural to me that there should be a language targeted at serious musicians which, provided the user is knowledgeable in the domain, is as easy to use as Melodica, but which provides more power and flexibility than a DAW. In short, I want to fill in the gap in this spectrum:
+
+Simple ---------------> Powerful
+Melodica -> DAWs -> ??? -> CSoud
 
 ### Domain
 _Describe the project's domain in five words._
 
+Instrument oriented digital music synthesis.
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain?_ 
 
+There are two parts to programming. The first is defining instruments. This is
+done by laying out a chain of sources and effects. These chains can be composed
+by connecting the output of one chain to the input of another (the second chain
+would have to start with an effect, which takes a signal as input). Chains can
+also be connected in parallel using a component that takes multiple sounds as
+input and produces a single sound as output.
+
+The second part of programming is writing a score. A score uses various
+instruments to produce a piece of music by playing a certain note on a certain
+instrument at a certain time. Playing a note means giving the instrument any
+inputs that it requires to start generating sound, such as pitch and amplitude.
+There's some flexibility here: the score-creation process could be a textual
+language that comes bundled with the instrument-creation language. This is what
+CSound does. Or, the instrument language could be designed to interface with
+existing methods for writing musical scores. For example, there could be a way
+to assign each track in a MIDI file to a different instrument, and use the MIDI
+messages like note-on, note-off, velocity and so on to pass parameters to the
+instruments.
```

Comment:
I think the first part to the programming is essential to what you want. I can understand why you would want the second half. However, I am curious if there is a way to play our new instruments besides writing a score (like connecting it to a midi keyboard).